portal-kernel/src/com/liferay/portal/kernel/util/SubscriptionSender.java

```diff
@@ -859,21 +942,41 @@ protected void sendUserNotification(User user) throws Exception {
 				_notificationType,
 				UserNotificationDeliveryConstants.TYPE_PUSH)) {
 
-			UserNotificationEventLocalServiceUtil.sendUserNotificationEvents(
-				user.getUserId(), portletId,
-				UserNotificationDeliveryConstants.TYPE_PUSH,
-				notificationEventJSONObject);
+			if (notifyImmediately) {
```

Comment:
Instead of the conditional, why not pass the "notifiyImmediately" value to the method?